Builds/CMake/CMakeFuncs.cmake

```diff
@@ -105,16 +105,36 @@ macro(parse_target)
       endif()
 
     endwhile()
+    set(CMAKE_C_COMPILER ${CMAKE_C_COMPILER} CACHE FILEPATH
```

Comment:
So I'm clear, the reason these variables are forcibly `set` into the cache, then `unset`, is because the `project` macro was otherwise overriding them?  What does the `unset` call do?

It might be worth leaving a  comment beyond the updated commit message to explain what is going on.